plugin.py

```diff
@@ -1355,6 +1355,17 @@ def __init__(self, irc):
         if self.registryValue('announceNagInterval') > 0:
             schedule.addEvent(self.checkNag,time.time()+self.registryValue('announceNagInterval'))
 
+    # helper for hop-support, orcus: 
+    # returns True if the bot owns the required mode
+    # Evaluates new 'allowBotBeingHOPorOP' option to determine the required mode minimal mode
+    # False: previous/old style in effect - requiring the bot to be OP
+    # True: alternative/new style in effect - requireing the bot to be HOP or OP
+    def botHasSufficientMode(self, theBotNick, theChannel):
+        if self.registryValue('allowBotBeingHOPorOP'):
+            return theBotNick in theChannel.ops or theBotNick in theChannel.halfops
+        else:
+            return theBotNick in theChannel.ops
```

Comment:
Note: Limnoria has `irc.state.channels[channel].isHalfopPlus(nick)` which is an easier way to write this.

For style consistency it's probably better to not include "the" in your variable names. i.e. `channel` instead of `theChannel`